File: Leetcode/26-removeDuplicatesFromSortedArray/removeDuplicatesFromSortedArray.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        l = 1
        for r in range(1, len(nums)):
            if nums[r] != nums[r-1]:
                nums[l] = nums[r]
                l+=1
        return l

# nums = [0,0,1,1,1,2,2,3,3,4]
# Nums is an array that is sorted in ascending order [1,2,3,4]
# Remove duplicates in-place sucht that each unique element appears only once
# The relative order of elements should be kept the same 
# Return the number of unique elements in nums, consider this be k 
# Change the array nums such that the first k elements of nums contain the unique elements in the order they were present in nums initial
# return k

# Example:
# Input: nums = [0,0,1,1,1,2,2,3,3,4]

File: Misc/Leetcode/26-removeDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        int l = 1;
        for (int r = 1; r < nums.Length; r++) {
            if (nums[r] != nums[r-1]) {
                nums[l] = nums[r];
                l++;
            }
        }
        return l;
    }
}

// Example usage:
// int[] nums = new int[] {0,0,1,1,1,2,2,3,3,4};
// The array is sorted in ascending order
// Remove duplicates in-place such that each unique element appears only once
// The relative order of elements should be kept the same
// Return the number of unique elements in nums (k)
// The first k elements of nums will contain the unique elements in their original order 